uriget/uriget.go

```diff
@@ -233,3 +239,38 @@ func (o *options) getGit(ctx context.Context, u *url.URL) ([]byte, error) {
 	o.logger.Printf("Read %d bytes from %s", len(buff), filepath.Join(td, subPath))
 	return buff, nil
 }
+
+func (o *options) getOci(ctx context.Context, u *url.URL) ([]byte, error) {
+	parts := strings.Split(u.Host+u.Path, "/")
+	if len(parts) < 2 {
+		return nil, fmt.Errorf("invalid OCI URL format")
+	}
+	registry := parts[0]
+	repo := strings.Join(parts[1:len(parts)-1], "/")
+	tag := "latest"
+	lastPart := parts[len(parts)-1]
+	if strings.Contains(lastPart, ":") {
+		split := strings.Split(lastPart, ":")
+		repo = strings.Join(parts[1:len(parts)-1], "/") + "/" + split[0]
+		tag = split[1]
+	}
+	store, err := oci.New(o.tempDir)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create OCI layout store: %w", err)
+	}
+	repoUrl := fmt.Sprintf("%s/%s", registry, repo)
```

Comment:
If we convert the url into a [Reference](https://pkg.go.dev/oras.land/oras-go/v2/registry#Reference), we don't need to compose the repo url anymore, but we can leverage [String](https://pkg.go.dev/oras.land/oras-go/v2/registry#Reference.String) method from it (`registry` and `repo` vars will not be needed anymore).